### PR TITLE
Showing cities on map

### PR DESCRIPTION
**Pull request by krsoninikhil**
Adding cities-on-map.html to show all the cities on map and giving link for the same in footer. (on #51 )

### CODE REVIEW

File: cities-on-map.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+layout: default
+---
+<script src="https://maps.googleapis.com/maps/api/js"></script>
```
Comment (captn3m0) [inline cities-on-map.html]: Can we use http://leafletjs.com/ instead of Google Maps?

File: _includes/footer.html [hunk 2]
```diff
@@ -8,6 +8,7 @@
         <ul class="contact-list">
           <li>{{ site.title }}</li>
           <li>{{ site.couches | size }} Hackers, {{site.couches|group_by:'country'|size}} Countries</li>
+          <li><a href="/cities-on-map.html">View cities on map</a></li>
```
Comment (captn3m0) [inline _includes/footer.html]: /map.html might be a better name

**Comment by captn3m0** [comment]
Thanks :+1:

File: cities-on-map.html (hunk 1 above)
Comment (krsoninikhil) [inline cities-on-map.html]: Sure, but is there any copyright issues with Google Maps?

File: _includes/footer.html (hunk 2 above)
Comment (krsoninikhil) [inline _includes/footer.html]: I'll change that.

**Comment by krsoninikhil** [comment]
My pleasure! :)
my-pleasure

File: cities-on-map.html (hunk 1 above)
Comment (captn3m0) [inline cities-on-map.html]: Not copyright, but tracking issues. I decided against linking to google maps earlier when someone told about nominatim. I think its better if we stick to open projects.

Comment (krsoninikhil) [inline cities-on-map.html]: Got it. Have a look on new commit with these changes.

**Comment by simison** [comment]
Can anyone think of any way to cache these things either at the browser or as a post-git hook or something when new couches are created? I feel bad hitting Nominatim so many times each time someone opens this page.

Does it send locations to Nominatim as a batch request or one by one?

https://wiki.openstreetmap.org/wiki/Nominatim_usage_policy

+1 for Leaflet
+1 thanks a lot!

**Comment by krsoninikhil** [comment]
Its one by one in a loop. 
If there is any way to create `.yml` file with the data obtained then there wont be any problem but I couldn't find a way to do so. Any help on that?

**Comment by simison** [comment]
@krsoninikhil yea there's a way, but others might know better the details how the best go about it: https://jekyllrb.com/docs/continuous-integration/

Giving Leaflet a [Geojson](http://leafletjs.com/reference.html#geojson) object might be the simplest way.

**Comment by captn3m0** [comment]
So, we need a way to generate the geojson files when our build runs, right? I'll work on a script later tonight for this. @krsoninikhil Will give you commit access to the repo as well, so we can work on the same branch.

**Comment by krsoninikhil** [comment]
I have written a shell script to generate the geojson file and that `json` file is used to put marker on map. This solve the issue of hitting Nominatim every time this is opened, but I'm not sure how to make the script run whenever a `jekyll build` happens (the currently present `/geo.json` file is generated by directly executing the `/_script/geojason.sh` in local machine).

Please have a look @captn3m0 @simison

**Comment by captn3m0** [comment]
@krsoninikhil I've added you as a collaborator to the repo. I'm pushing a branch called `cities-on-map`. Lets work on that instead. My plan is to write a Jekyll plugin that we can call in a custom layout for this.

**Comment by krsoninikhil** [comment]
Okay. So this way a ruby script i.e. the plugin will generate a geojson file?
Isn't there a way to run a shell script before site generating?

**Comment by krsoninikhil** [comment]
@captn3m0 I have written a generator plugin and push in master. Its working fine, please check it.

**Comment by captn3m0** [comment]
Looks mostly good. I'm moving it to a more jekyll like approach, though.

**Comment by captn3m0** [comment]
Closing this in favor of #84, which is based on a local branch. @krsoninikhil has commit rights so we can both work on it much more easily now.

**Pull request closed without merging**